### ISSUE TITLE
toString generates invalid code

### ISSUE THREAD

**Issue by iTakeshi** [scalameta/scalameta#572]
Hi, 

```scala
val tree=
  Term.Apply(
    Term.Name("foo"),
    Seq(
      Term.Function(
        Seq(Term.Param(Seq(Mod.Implicit()), Term.Name("i"), Some(Type.Name("Int")), None)),
        Lit(())
      )
    )
  )
println(tree.toString)
```
will be `foo(implicit i: Int => ())`. This code results in a compile error.
```
<console>:1: '=>' expected but ')' found.
foo(implicit i: Int => ())
                         ^
```

This should be printed as `foo { implicit i: Int => () }`.

**Comment by olafurpg** [comment]
The pretty printer contains a ton of bugs, see https://github.com/scalameta/scalameta/issues/567#issuecomment-267061270. I suspect it's not too hard to fix most of these, it boils down to finding the right branch in TreeSyntax (e.g., [here](https://github.com/scalameta/scalameta/blob/ff054d55f28c63c860f56fea74ae67adbc7fc791/scalameta/trees/src/main/scala/scala/meta/internal/prettyprinters/TreeSyntax.scala#L234)) and then add a test in syntax suite like [here](https://github.com/scalameta/scalameta/blob/ff054d55f28c63c860f56fea74ae67adbc7fc791/scalameta/scalameta/src/test/scala/scala/meta/tests/prettyprinters/SyntacticSuite.scala#L470-L473).

**Comment by iTakeshi** [comment]
I'm just inspecting around there and found the cause, preparing PR. Thanks @olafurpg

**Comment by iTakeshi** [comment]
solved by #574
